### PR TITLE
Favorite/unfavorite articles on the homepage + plus other miscellaneous

### LINKED ISSUE #46: [Home page] Favoriting/unfavoriting
We should have the ability to favorite/unfavorite articles

Features:
* Unfavorited state
* Favorited state
* Click to favorite
* Click to unfavorite
* Count should increment/decrement when you favorite/unfavorite

### Unfavorited
image

### Favorited
image

Want to write tests?
* Integration: Click to favorite, state change, count increments
* Integration: Click to unfavorite, state change, count increments

### PR DESCRIPTION
**Pull request by patocallaghan**
Fixes #46 

Adds the ability to favorite/unfavorite posts and some extra miscellaneous things

* Split author meta information out into its own component, `article-author`
* Split favorite/unfavorite logic out into its own component, `favorite-article`
* Removed lots of empty and unnecessary files i.e. .gitkeep and empty route/controller/component files
* Remove moment and related addons as it was only used in a single location. Swapped it our for using a custom helper `format-date`
  * before: `766.66 KB (192.78 KB gzipped)`
  * after: `681.42 KB (171.74 KB gzipped)`)
* Follow up to a previous conversation in a previous PR and fixed resetting the feed query param in link-tos

favoriting

### CODE REVIEW

File: app/models/article.js [hunk 1]
```diff
@@ -1,63 +1,35 @@
 import DS from 'ember-data';
+import { inject as service } from '@ember/service';
 
 const { attr, belongsTo, hasMany } = DS;
 
 export default DS.Model.extend({
-  /**
```
Comment (patocallaghan) [inline app/models/article.js]: Removed comments as they don't appear to give much value. Also I don't set a default `date` as those values get set by the server.

File: app/templates/components/feed-item.hbs [hunk 2]
```diff
@@ -1,27 +1,18 @@
...
-    </div>
-    <button class="btn {{if article.favorited "btn-primary" "btn-outline-primary"}} btn-sm pull-xs-right"
-      data-test-article-favoritesCount>
-      <i class="ion-heart"></i> {{article.favoritesCount}}
-    </button>
+    {{article-author article=article}}
+    {{favorite-article article=article isIconOnly=true class="pull-xs-right"}}
   </div>
-  {{#link-to "article" article.slug class="preview-link" }}
+  {{#link-to "article" article.id class="preview-link" }}
```
Comment (patocallaghan) [inline app/templates/components/feed-item.hbs]: Fixes linking to the wrong location

File: app/templates/components/feed-item.hbs [hunk 3]
```diff
@@ -1,27 +1,18 @@
...
     <h1 data-test-article-title>{{article.title}}</h1>
     <p data-test-article-description>{{article.description}}</p>
     <span>Read more...</span>
-    <ul class="tag-list">
-      {{#each article.tagList as |tag|}}
-        <li class="tag-default tag-pill tag-outline" data-test-article-tag>{{tag.value}}</li>
-      {{/each}}
-    </ul>
+    {{#if article.tagList}}
```
Comment (patocallaghan) [inline app/templates/components/feed-item.hbs]: Add an extra if to only render this list if the tagList exists

File: app/templates/components/favorite-article.hbs [hunk 4]
```diff
@@ -0,0 +1,10 @@
+<i class="ion-heart"></i>
+&nbsp;
+{{#if @isIconOnly}}
```
Comment (Alonski) [inline app/templates/components/favorite-article.hbs]: Some Octane snuck into here :)

File: app/templates/components/footer-bar.hbs [hunk 5]
```diff
@@ -1,8 +1,9 @@
 <div class="container">
-  {{#link-to "home"  (query-params author=null tag=null page=1) class="logo-font" }}
+  {{#link-to "home"  (query-params feed=null tag=null page=1) class="logo-font" }}
     conduit
   {{/link-to}}
   <span class="attribution">
-    An interactive learning project from <a href="https://thinkster.io">Thinkster</a>. Code &amp; design licensed under MIT.
+    An interactive learning project from <a href="https://thinkster.io">Thinkster</a>. Code &amp; design licensed under
```
Comment (Alonski) [inline app/templates/components/footer-bar.hbs]: Was this changed by accident?

Comment (patocallaghan) [reply app/templates/components/footer-bar.hbs]: This looks to be Prettier doing its thing, saving the file triggers this layout.

File: app/templates/components/favorite-article.hbs (hunk 4 above)
Comment (patocallaghan) [reply app/templates/components/favorite-article.hbs]: Good catch, updated

**Pull request merged**